project/build.scala

```diff
@@ -106,6 +106,7 @@ object build {
   )
 
   private val Scala211_jvm_and_js_options = Seq(
+    "-Xexperimental", // SAM types
```

Comment:
Nice.

Reply:
> I don't understand what the different entries in the build matrix are. @xuwei-k why is one 2.11.11 / java8 build working but another is failing?

It's scala-native build. Maybe scala-native does not support "SAM types".
Please use 
`Equal.equal { (z1, z2) =>` 
instead of
 `{ (z1, z2) =>`

Reply:
oh! `Scala211_jvm_and_js_options` of course there is probably another variable somewhere for the native build.